<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_20) on Thu Aug 12 14:27:43 EDT 2010 -->
<META http-equiv="Content-Type" content="text/html; charset=UTF-8">
<TITLE>
RIO
</TITLE>

<META NAME="date" CONTENT="2010-08-12">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../../../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="RIO";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="class-use/RIO.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="../../../org/forester/sdi/ORcount.html" title="class in org.forester.sdi"><B>PREV CLASS</B></A>&nbsp;
&nbsp;<A HREF="../../../org/forester/sdi/RIOn.html" title="class in org.forester.sdi"><B>NEXT CLASS</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../index.html?org/forester/sdi/RIO.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="RIO.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;NESTED&nbsp;|&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<HR>
<!-- ======== START OF CLASS DATA ======== -->
<H2>
<FONT SIZE="-1">
org.forester.sdi</FONT>
<BR>
Class RIO</H2>
<PRE>
java.lang.Object
  <IMG SRC="../../../resources/inherit.gif" ALT="extended by "><B>org.forester.sdi.RIO</B>
</PRE>
<HR>
<DL>
<DT><PRE>public final class <B>RIO</B><DT>extends java.lang.Object</DL>
</PRE>

<P>
<HR>

<P>

<!-- ======== CONSTRUCTOR SUMMARY ======== -->

<A NAME="constructor_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Constructor Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../org/forester/sdi/RIO.html#RIO()">RIO</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Default constructor.</TD>
</TR>
</TABLE>
&nbsp;
<!-- ========== METHOD SUMMARY =========== -->

<A NAME="method_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Method Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../org/forester/sdi/RIO.html#getBootstraps()">getBootstraps</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns the numbers of trees analyzed.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../org/forester/sdi/RIO.html#getDistance(java.lang.String)">getDistance</A></B>(java.lang.String&nbsp;name)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns the distance to a sequences/taxa after a distance list file has
 been read in with readDistanceList(File).</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../org/forester/sdi/RIO.html#getDistance(java.lang.String, java.lang.String)">getDistance</A></B>(java.lang.String&nbsp;name1,
            java.lang.String&nbsp;name2)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../org/forester/sdi/RIO.html#getExtNodesOfAnalyzedGeneTrees()">getExtNodesOfAnalyzedGeneTrees</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns the numbers of number of ext nodes in gene trees analyzed (after
 stripping).</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.HashMap&lt;java.lang.String,java.lang.Integer&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../org/forester/sdi/RIO.html#getInferredOrthologs(java.lang.String)">getInferredOrthologs</A></B>(java.lang.String&nbsp;seq_name)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns a HashMap containing the inferred orthologs of the external gene
 tree node with the sequence name seq_name.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.HashMap&lt;java.lang.String,java.lang.Integer&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../org/forester/sdi/RIO.html#getInferredSuperOrthologs(java.lang.String)">getInferredSuperOrthologs</A></B>(java.lang.String&nbsp;seq_name)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns a HashMap containing the inferred "super orthologs" of the
 external gene tree node with the sequence name seq_name.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.HashMap&lt;java.lang.String,java.lang.Integer&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../org/forester/sdi/RIO.html#getInferredUltraParalogs(java.lang.String)">getInferredUltraParalogs</A></B>(java.lang.String&nbsp;seq_name)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns a HashMap containing the inferred "ultra paralogs" of the
 external gene tree node with the sequence name seq_name.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../org/forester/sdi/RIO.html#getOrder(int)">getOrder</A></B>(int&nbsp;sort)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns the order in which ortholog (o), "super ortholog" (s) and
 distance (d) are returned and sorted (priority of sort always goes from
 left to right), given sort.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.StringBuffer</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../org/forester/sdi/RIO.html#getOrderHelp()">getOrderHelp</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;long</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../org/forester/sdi/RIO.html#getTime()">getTime</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns the time (in ms) needed to run "inferOrthologs".</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../org/forester/sdi/RIO.html#inferOrthologs(java.io.File, org.forester.phylogeny.Phylogeny, java.lang.String)">inferOrthologs</A></B>(java.io.File&nbsp;gene_trees_file,
               <A HREF="../../../org/forester/phylogeny/Phylogeny.html" title="class in org.forester.phylogeny">Phylogeny</A>&nbsp;species_tree,
               java.lang.String&nbsp;query)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Infers the orthologs (as well the "super orthologs", the "subtree
 neighbors", and the "ultra paralogs") for each external node of the gene
 Trees in multiple tree File gene_trees_file (=output of PHYLIP NEIGHBOR,
 for example).</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.ArrayList&lt;java.lang.String&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../org/forester/sdi/RIO.html#inferredOrthologsToArrayList(java.lang.String, double)">inferredOrthologsToArrayList</A></B>(java.lang.String&nbsp;seq_name,
                             double&nbsp;threshold_orthologs)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns an ArrayList containg the names of orthologs of the PhylogenyNode
 with seq name seq_name.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.StringBuffer</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../org/forester/sdi/RIO.html#inferredOrthologsToString(java.lang.String, int, double, double)">inferredOrthologsToString</A></B>(java.lang.String&nbsp;query_name,
                          int&nbsp;sort,
                          double&nbsp;threshold_orthologs,
                          double&nbsp;threshold_subtreeneighborings)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns a String containg the names of orthologs of the PhylogenyNode
 with seq name query_name.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../org/forester/sdi/RIO.html#inferredOrthologTableToFile(java.io.File)">inferredOrthologTableToFile</A></B>(java.io.File&nbsp;outfile)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Writes the orthologs for each external node of the gene trees to outfile
 in the form of a table.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../org/forester/sdi/RIO.html#inferredSuperOrthologTableToFile(java.io.File)">inferredSuperOrthologTableToFile</A></B>(java.io.File&nbsp;outfile)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Writes the "super orthologs" for each external nodes of the gene trees to
 outfile in the form of a table.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../org/forester/sdi/RIO.html#inferredUltraParalogsToString(java.lang.String, boolean, double)">inferredUltraParalogsToString</A></B>(java.lang.String&nbsp;query_name,
                              boolean&nbsp;return_dists,
                              double&nbsp;threshold_ultra_paralogs)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns a String containg the names of orthologs of the PhylogenyNode
 with seq name query_name.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../org/forester/sdi/RIO.html#readDistanceMatrix(java.io.File)">readDistanceMatrix</A></B>(java.io.File&nbsp;matrix_file)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;<A NAME="methods_inherited_from_class_java.lang.Object"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left"><B>Methods inherited from class java.lang.Object</B></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE>clone, equals, finalize, getClass, hashCode, notify, notifyAll, toString, wait, wait, wait</CODE></TD>
</TR>
</TABLE>
&nbsp;
<P>

<!-- ========= CONSTRUCTOR DETAIL ======== -->

<A NAME="constructor_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Constructor Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="RIO()"><!-- --></A><H3>
RIO</H3>
<PRE>
public <B>RIO</B>()</PRE>
<DL>
<DD>Default constructor.
<P>
</DL>

<!-- ============ METHOD DETAIL ========== -->

<A NAME="method_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Method Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="getBootstraps()"><!-- --></A><H3>
getBootstraps</H3>
<PRE>
public final int <B>getBootstraps</B>()</PRE>
<DL>
<DD>Returns the numbers of trees analyzed.
<P>
<DD><DL>

<DT><B>Returns:</B><DD>the numbers of trees analyzed</DL>
</DD>
</DL>
<HR>

<A NAME="getDistance(java.lang.String)"><!-- --></A><H3>
getDistance</H3>
<PRE>
public final double <B>getDistance</B>(java.lang.String&nbsp;name)</PRE>
<DL>
<DD>Returns the distance to a sequences/taxa after a distance list file has
 been read in with readDistanceList(File). Throws an exception if name is
 not found or if no list has been read in.
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>name</CODE> - a sequence name</DL>
</DD>
</DL>
<HR>

<A NAME="getDistance(java.lang.String, java.lang.String)"><!-- --></A><H3>
getDistance</H3>
<PRE>
public final double <B>getDistance</B>(java.lang.String&nbsp;name1,
                                java.lang.String&nbsp;name2)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getExtNodesOfAnalyzedGeneTrees()"><!-- --></A><H3>
getExtNodesOfAnalyzedGeneTrees</H3>
<PRE>
public final int <B>getExtNodesOfAnalyzedGeneTrees</B>()</PRE>
<DL>
<DD>Returns the numbers of number of ext nodes in gene trees analyzed (after
 stripping).
<P>
<DD><DL>

<DT><B>Returns:</B><DD>number of ext nodes in gene trees analyzed (after stripping)</DL>
</DD>
</DL>
<HR>

<A NAME="getInferredOrthologs(java.lang.String)"><!-- --></A><H3>
getInferredOrthologs</H3>
<PRE>
public final java.util.HashMap&lt;java.lang.String,java.lang.Integer&gt; <B>getInferredOrthologs</B>(java.lang.String&nbsp;seq_name)</PRE>
<DL>
<DD>Returns a HashMap containing the inferred orthologs of the external gene
 tree node with the sequence name seq_name. Sequence names are the keys
 (String), numbers of observations are the values (Int). Orthologs are to
 be inferred by method "inferOrthologs". Throws an exception if seq_name
 is not found.
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>seq_name</CODE> - sequence name of a external node of the gene trees
<DT><B>Returns:</B><DD>HashMap containing the inferred orthologs
         (name(String)->value(Int))</DL>
</DD>
</DL>
<HR>

<A NAME="getInferredSuperOrthologs(java.lang.String)"><!-- --></A><H3>
getInferredSuperOrthologs</H3>
<PRE>
public final java.util.HashMap&lt;java.lang.String,java.lang.Integer&gt; <B>getInferredSuperOrthologs</B>(java.lang.String&nbsp;seq_name)</PRE>
<DL>
<DD>Returns a HashMap containing the inferred "super orthologs" of the
 external gene tree node with the sequence name seq_name. Sequence names
 are the keys (String), numbers of observations are the values (Int).
 Super orthologs are to be inferred by method "inferOrthologs". Throws an
 exception if seq_name is not found.
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>seq_name</CODE> - sequence name of a external node of the gene trees
<DT><B>Returns:</B><DD>HashMap containing the inferred super orthologs
         (name(String)->value(Int))</DL>
</DD>
</DL>
<HR>

<A NAME="getInferredUltraParalogs(java.lang.String)"><!-- --></A><H3>
getInferredUltraParalogs</H3>
<PRE>
public final java.util.HashMap&lt;java.lang.String,java.lang.Integer&gt; <B>getInferredUltraParalogs</B>(java.lang.String&nbsp;seq_name)</PRE>
<DL>
<DD>Returns a HashMap containing the inferred "ultra paralogs" of the
 external gene tree node with the sequence name seq_name. Sequence names
 are the keys (String), numbers of observations are the values (Int).
 "ultra paralogs" are to be inferred by method "inferOrthologs". Throws an
 exception if seq_name is not found.
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>seq_name</CODE> - sequence name of a external node of the gene trees
<DT><B>Returns:</B><DD>HashMap containing the inferred ultra paralogs
         (name(String)->value(Int))</DL>
</DD>
</DL>
<HR>

<A NAME="getTime()"><!-- --></A><H3>
getTime</H3>
<PRE>
public long <B>getTime</B>()</PRE>
<DL>
<DD>Returns the time (in ms) needed to run "inferOrthologs". Final variable
 TIME needs to be set to true.
<P>
<DD><DL>

<DT><B>Returns:</B><DD>time (in ms) needed to run method "inferOrthologs"</DL>
</DD>
</DL>
<HR>

<A NAME="inferOrthologs(java.io.File, org.forester.phylogeny.Phylogeny, java.lang.String)"><!-- --></A><H3>
inferOrthologs</H3>
<PRE>
public void <B>inferOrthologs</B>(java.io.File&nbsp;gene_trees_file,
                           <A HREF="../../../org/forester/phylogeny/Phylogeny.html" title="class in org.forester.phylogeny">Phylogeny</A>&nbsp;species_tree,
                           java.lang.String&nbsp;query)
                    throws java.io.IOException</PRE>
<DL>
<DD>Infers the orthologs (as well the "super orthologs", the "subtree
 neighbors", and the "ultra paralogs") for each external node of the gene
 Trees in multiple tree File gene_trees_file (=output of PHYLIP NEIGHBOR,
 for example). Tallies how many times each sequence is (super-)
 orthologous towards the query. Tallies how many times each sequence is
 ultra paralogous towards the query. Tallies how many times each sequence
 is a subtree neighbor of the query. Gene duplications are inferred using
 SDI. Modifies its argument species_tree. Is a little faster than
 "inferOrthologs(File,Phylogeny)" since orthologs are only inferred for
 query.
 <p>
 To obtain the results use the methods listed below.
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>gene_trees_file</CODE> - a File containing gene Trees in NH format, which is the result
            of performing a bootstrap analysis in PHYLIP<DD><CODE>species_tree</CODE> - a species Phylogeny, which has species names in its species
            fields<DD><CODE>query</CODE> - the sequence name of the squence whose orthologs are to be
            inferred
<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="inferredOrthologsToArrayList(java.lang.String, double)"><!-- --></A><H3>
inferredOrthologsToArrayList</H3>
<PRE>
public java.util.ArrayList&lt;java.lang.String&gt; <B>inferredOrthologsToArrayList</B>(java.lang.String&nbsp;seq_name,
                                                                          double&nbsp;threshold_orthologs)</PRE>
<DL>
<DD>Returns an ArrayList containg the names of orthologs of the PhylogenyNode
 with seq name seq_name.
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>seq_name</CODE> - sequence name of a external node of the gene trees<DD><CODE>threshold_orthologs</CODE> - the minimal number of observations for a a sequence to be
            reported as orthologous as percentage (0.0-100.0%)
<DT><B>Returns:</B><DD>ArrayList containg the names of orthologs of the PhylogenyNode
         with seq name seq_name</DL>
</DD>
</DL>
<HR>

<A NAME="inferredOrthologsToString(java.lang.String, int, double, double)"><!-- --></A><H3>
inferredOrthologsToString</H3>
<PRE>
public java.lang.StringBuffer <B>inferredOrthologsToString</B>(java.lang.String&nbsp;query_name,
                                                        int&nbsp;sort,
                                                        double&nbsp;threshold_orthologs,
                                                        double&nbsp;threshold_subtreeneighborings)</PRE>
<DL>
<DD>Returns a String containg the names of orthologs of the PhylogenyNode
 with seq name query_name. The String also contains how many times a
 particular ortholog has been observed.
 <p>
 <ul>
 The output order is (per line): Name, Ortholog, Subtree neighbor, Super
 ortholog, Distance
 </ul>
 <p>
 The sort priority of this is determined by sort in the following manner:
 <ul>
 <li>0 : Ortholog
 <li>1 : Ortholog, Super ortholog
 <li>2 : Super ortholog, Ortholog
 <li>3 : Ortholog, Distance
 <li>4 : Distance, Ortholog
 <li>5 : Ortholog, Super ortholog, Distance
 <li>6 : Ortholog, Distance, Super ortholog
 <li>7 : Super ortholog, Ortholog, Distance
 <li>8 : Super ortholog, Distance, Ortholog
 <li>9 : Distance, Ortholog, Super ortholog
 <li>10 : Distance, Super ortholog, Ortholog
 <li>11 : Ortholog, Subtree neighbor, Distance
 <li>12 : Ortholog, Subtree neighbor, Super ortholog, Distance (default)
 <li>13 : Ortholog, Super ortholog, Subtree neighbor, Distance
 <li>14 : Subtree neighbor, Ortholog, Super ortholog, Distance
 <li>15 : Subtree neighbor, Distance, Ortholog, Super ortholog
 <li>16 : Ortholog, Distance, Subtree neighbor, Super ortholog
 <li>17 : Ortholog, Subtree neighbor, Distance, Super ortholog
 </ul>
 <p>
 Returns "-" if no putative orthologs have been found (given
 threshold_orthologs).
 <p>
 Orthologs are to be inferred by method "inferOrthologs".
 <p>
 (Last modified: 05/08/01)
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>query_name</CODE> - sequence name of a external node of the gene trees<DD><CODE>sort</CODE> - order and sort priority<DD><CODE>threshold_orthologs</CODE> - the minimal number of observations for a a sequence to be
            reported as orthologous, in percents (0.0-100.0%)<DD><CODE>threshold_subtreeneighborings</CODE> - the minimal number of observations for a a sequence to be
            reported as orthologous, in percents (0.0-100.0%)
<DT><B>Returns:</B><DD>String containing the inferred orthologs, String containing "-"
         if no orthologs have been found null in case of error<DT><B>See Also:</B><DD><A HREF="../../../org/forester/sdi/RIO.html#inferOrthologs(java.io.File, org.forester.phylogeny.Phylogeny, java.lang.String)"><CODE>inferOrthologs(File,Phylogeny,String)</CODE></A>, 
<CODE>#inferOrthologs(Phylogeny[],Phylogeny)</CODE>, 
<CODE>#inferOrthologs(File,Phylogeny)</CODE>, 
<A HREF="../../../org/forester/sdi/RIO.html#getOrder(int)"><CODE>getOrder(int)</CODE></A></DL>
</DD>
</DL>
<HR>

<A NAME="inferredOrthologTableToFile(java.io.File)"><!-- --></A><H3>
inferredOrthologTableToFile</H3>
<PRE>
public void <B>inferredOrthologTableToFile</B>(java.io.File&nbsp;outfile)
                                 throws java.io.IOException</PRE>
<DL>
<DD>Writes the orthologs for each external node of the gene trees to outfile
 in the form of a table. Orthologs are to be inferred by method
 "inferOrthologs". Overwrites without asking! (Last modified: 12/07/00)
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>outfile</CODE> - the File to write to
<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="inferredSuperOrthologTableToFile(java.io.File)"><!-- --></A><H3>
inferredSuperOrthologTableToFile</H3>
<PRE>
public void <B>inferredSuperOrthologTableToFile</B>(java.io.File&nbsp;outfile)
                                      throws java.io.IOException</PRE>
<DL>
<DD>Writes the "super orthologs" for each external nodes of the gene trees to
 outfile in the form of a table. Super orthologs are to be inferred by
 method "inferOrthologs". Overwrites without asking!
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>outfile</CODE> - the File to write to
<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="inferredUltraParalogsToString(java.lang.String, boolean, double)"><!-- --></A><H3>
inferredUltraParalogsToString</H3>
<PRE>
public java.lang.String <B>inferredUltraParalogsToString</B>(java.lang.String&nbsp;query_name,
                                                      boolean&nbsp;return_dists,
                                                      double&nbsp;threshold_ultra_paralogs)</PRE>
<DL>
<DD>Returns a String containg the names of orthologs of the PhylogenyNode
 with seq name query_name. The String also contains how many times a
 particular ortholog has been observed. Returns "-" if no putative
 orthologs have been found (given threshold_orthologs).
 <p>
 Orthologs are to be inferred by method "inferOrthologs".
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>query_name</CODE> - sequence name of a external node of the gene trees<DD><CODE>return_dists</CODE> - <DD><CODE>threshold_ultra_paralogs</CODE> - between 1 and 100
<DT><B>Returns:</B><DD>String containing the inferred orthologs, String containing "-"
         if no orthologs have been found null in case of error</DL>
</DD>
</DL>
<HR>

<A NAME="readDistanceMatrix(java.io.File)"><!-- --></A><H3>
readDistanceMatrix</H3>
<PRE>
public final void <B>readDistanceMatrix</B>(java.io.File&nbsp;matrix_file)
                              throws java.io.IOException</PRE>
<DL>
<DD><DL>

<DT><B>Throws:</B>
<DD><CODE>java.io.IOException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="getOrder(int)"><!-- --></A><H3>
getOrder</H3>
<PRE>
public static final java.lang.String <B>getOrder</B>(int&nbsp;sort)</PRE>
<DL>
<DD>Returns the order in which ortholog (o), "super ortholog" (s) and
 distance (d) are returned and sorted (priority of sort always goes from
 left to right), given sort. For the meaning of sort
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>sort</CODE> - determines order and sort priority
<DT><B>Returns:</B><DD>String indicating the order<DT><B>See Also:</B><DD><A HREF="../../../org/forester/sdi/RIO.html#inferredOrthologsToString(java.lang.String, int, double, double)"><CODE>inferredOrthologsToString(String,int,double,double)</CODE></A></DL>
</DD>
</DL>
<HR>

<A NAME="getOrderHelp()"><!-- --></A><H3>
getOrderHelp</H3>
<PRE>
public static final java.lang.StringBuffer <B>getOrderHelp</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<!-- ========= END OF CLASS DATA ========= -->
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="class-use/RIO.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="../../../org/forester/sdi/ORcount.html" title="class in org.forester.sdi"><B>PREV CLASS</B></A>&nbsp;
&nbsp;<A HREF="../../../org/forester/sdi/RIOn.html" title="class in org.forester.sdi"><B>NEXT CLASS</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../index.html?org/forester/sdi/RIO.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="RIO.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;NESTED&nbsp;|&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<HR>

</BODY>
</HTML>
